Lib/glyphsLib/__init__.py

```diff
@@ -72,35 +78,55 @@ def build_masters(filename, master_dir, designspace_instance_dir=None,
 
     Returns:
-        A list of master UFOs, and if designspace_instance_dir is provided, a
-        path to a designspace and a list of (path, data) tuples with instance
-        paths from the designspace and respective data from the Glyphs source.
+        A tuple of master UFOs, the path to the designspace file and a list of
+        (path, data) tuples with instance paths from the designspace and
+        respective data from the Glyphs source.
     """
 
     font = GSFont(filename)
-    instance_dir = None
-    if designspace_instance_dir is not None:
+
+    if designspace_instance_dir is None:
+        instance_dir = None
+    else:
         instance_dir = os.path.relpath(designspace_instance_dir, master_dir)
+
     designspace = to_designspace(
-        font, family_name=family_name, propagate_anchors=propagate_anchors,
-        instance_dir=instance_dir)
+        font,
+        family_name=family_name,
+        propagate_anchors=propagate_anchors,
+        instance_dir=instance_dir,
+        minimize_glyphs_diffs=minimize_glyphs_diffs)
+
     ufos = []
     for source in designspace.sources:
         ufos.append(source.font)
+
+        if minimize_glyphs_diffs:
+            ufo_create_background_layer_for_all_glyphs(source.font)
+
         ufo_path = os.path.join(master_dir, source.filename)
         clean_ufo(ufo_path)
         source.font.save(ufo_path)
 
-    if designspace_instance_dir is not None:
-        designspace_path = os.path.join(master_dir, designspace.filename)
-        designspace.write(designspace_path)
-        # All the instance data should be in the designspace. That's why for
-        # now we return the full designspace in place of `instance_data`.
-        # However, other functions still expect the instance data to have
-        # a list of (path, something) tuples, hence the class `InstanceData`.
-        return ufos, designspace_path, InstanceData(designspace)
-    else:
-        return ufos
+        if normalize_ufos:
+            try:
+                import ufonormalizer
+                ufonormalizer.normalizeUFO(ufo_path, writeModTimes=False)
+            except ImportError:
+                logger.warning("ufonormalizer not found, UFOs may contain "
+                               "more differences than necessary.")
+            except TypeError:
+                logger.warning("ufonormalizer found but too old, UFOs may "
+                               "contain more differences than necessary.")
+
+    designspace_path = os.path.join(master_dir, designspace.filename)
+    designspace.write(designspace_path)
+
+    # All the instance data should be in the designspace. That's why for
+    # now we return the full designspace in place of `instance_data`.
+    # However, other functions still expect the instance data to have
+    # a list of (path, something) tuples, hence the class `InstanceData`.
+    return ufos, designspace_path, InstanceData(designspace)
```

Comment:
note that fontmake (master) no longer uses this deprecated InstanceData list-of-tuples. You can drop it, unless it is being used by other internal glyphsLib functions.